pandas/core/arrays/datetimelike.py

```diff
@@ -919,7 +920,8 @@ def __add__(self, other):
             # DatetimeIndex, ndarray[datetime64]
             return self._add_datetime_arraylike(other)
         elif is_integer_dtype(other):
-            maybe_integer_op_deprecated(self)
+            if not is_period_dtype(self):
```

Comment:
This repeated if checks could maybe be moved into the `maybe_integer_op_deprecated`. @jbrockmendel is there a recommended Cython equivalent for `is_period_dtype`?